agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2142,6 +2364,154 @@ bool backhaul_manager::handle_1905_higher_layer_data_message(ieee1905_1::CmduMes
+     * that "If the value is 0, then the EUI48 field is not present; if the value is 1, then the
+     * EUI-48 field shall be present."
+     *
+     * However, optional fields are not currently supported by TLVF.
+     *
+     * As a workaround, instead of defining a tlvLinkMetricQuery TLV with an optional field, we
+     * have defined two different TLVs: tlvLinkMetricQuery with the optional field and
+     * tlvLinkMetricQueryAllNeighbors without it. Application must check which of both TLVs has
+     * been received inside the message.
+     */
+    std::shared_ptr<ieee1905_1::tlvLinkMetricQueryAllNeighbors> tlvLinkMetricQueryAllNeighbors;
+    std::shared_ptr<ieee1905_1::tlvLinkMetricQuery> tlvLinkMetricQuery;
+
+    tlvLinkMetricQueryAllNeighbors = cmdu_rx.getClass<ieee1905_1::tlvLinkMetricQueryAllNeighbors>();
+    if (!tlvLinkMetricQueryAllNeighbors) {
+        tlvLinkMetricQuery = cmdu_rx.getClass<ieee1905_1::tlvLinkMetricQuery>();
+        if (!tlvLinkMetricQuery) {
+            LOG(ERROR) << "getClass ieee1905_1::tlvLinkMetricQueryAllNeighbors and "
+                          "ieee1905_1::tlvLinkMetricQuery failed";
+            return false;
+        }
+    }
+
+    /**
+     * 1905.1 AL MAC address of the device that transmits the response message.
+     */
+    sMacAddr reporter_al_mac = network_utils::mac_from_string(bridge_info.mac);
+
+    /**
+     * 1905.1 AL MAC address of a neighbor of the receiving device.
+     * Query can specify a particular neighbor device or all neighbor devices.
+     */
+    sMacAddr neighbor_al_mac = network_utils::ZERO_MAC;
+
+    /**
+     * Obtain link metrics for either all neighbors or a specific neighbor
+     */
+    ieee1905_1::eLinkMetricNeighborType neighbor_type;
+
+    /**
+     * The link metrics type requested: TX, RX or both
+     */
+    ieee1905_1::eLinkMetricsType link_metrics_type;
+
+    if (tlvLinkMetricQuery) {
+        /**
+    	   * If tlvLinkMetricQuery has been included in message, we will be permissive enough to
+    	   * allow it specify ALL_NEIGHBORS and if so, then we will just ignore the field
+    	   * containing the MAC address of neighbor.
+    	   */
+        neighbor_type     = tlvLinkMetricQuery->neighbor_type();
+        neighbor_al_mac   = tlvLinkMetricQuery->mac_al_1905_device();
+        link_metrics_type = tlvLinkMetricQuery->link_metrics_type();
+    } else {
+        neighbor_type = tlvLinkMetricQueryAllNeighbors->neighbor_type();
+        if (ieee1905_1::eLinkMetricNeighborType::ALL_NEIGHBORS != neighbor_type) {
+            LOG(ERROR) << "Unexpected neighbor type: " << std::hex << int(neighbor_type);
+            return false;
+        }
+        link_metrics_type = tlvLinkMetricQueryAllNeighbors->link_metrics_type();
+    }
+
+    /**
+     * Set alias flag to true if link metrics for a specific neighbor have been requested
+     */
+    bool specific_neighbor =
+        ieee1905_1::eLinkMetricNeighborType::SPECIFIC_NEIGHBOR == neighbor_type;
```

Comment:
(nitpick) I would set this directly in the condition above instead of saving neighbor_type.